core/src/main/java/org/projectnessie/cel/common/types/IntT.java

```diff
@@ -216,6 +200,62 @@ public Val convertToType(Type typeValue) {
     return newTypeConversionError(IntType, typeValue);
   }
 
+  /** Compare implements traits.Comparer.Compare. */
+  @Override
+  public Val compare(Val other) {
+    switch (other.type().typeEnum()) {
+      case Double:
+        DoubleT cmp = (DoubleT) convertToType(DoubleType);
+        return cmp.compare(other);
+      case Uint:
+        if (i < 0L) {
+          // this int is negative, so it MUST be smaller than any uint
+          return IntNegOne;
+        }
+        if (other.intValue() < 0L) {
+          // the OTHER uint is > Integer.MAX_VALUE, so THIS int MUST be smaller
+          return IntNegOne;
+        }
+        return intOfCompare(Long.compareUnsigned(i, other.intValue()));
```

Comment:
nit: At this point the values are within the `[0; Integer.MAX_VALUE]` range... Do we really need to use `Long.compareUnsigned()`?